Modules/Loadable/VolumeRendering/Logic/vtkSlicerVolumeRenderingLogic.cxx

```diff
@@ -764,27 +768,48 @@ void vtkSlicerVolumeRenderingLogic::CopyLabelMapDisplayToVolumeRenderingDisplayN
 void vtkSlicerVolumeRenderingLogic::FitROIToVolume(vtkMRMLVolumeRenderingDisplayNode* vspNode)
 {
   // Resize the ROI to fit the volume
-  vtkMRMLAnnotationROINode *roiNode = vtkMRMLAnnotationROINode::SafeDownCast(vspNode->GetROINode());
   vtkMRMLScalarVolumeNode *volumeNode = vtkMRMLScalarVolumeNode::SafeDownCast(vspNode->GetVolumeNode());
 
-  if (volumeNode && roiNode)
-  {
-    int disabledModify = roiNode->StartModify();
+  if (!volumeNode)
+    {
+    return;
+    }
+
+  vtkMRMLAnnotationROINode* roiNode = vspNode->GetROINode();
+  vtkMRMLMarkupsROINode* markupsROINode = vspNode->GetMarkupsROINode();
+  if (roiNode)
+    {
+    MRMLNodeModifyBlocker blocker(roiNode);
 
     double xyz[3] = {0.0};
     double center[3] = {0.0};
 
     vtkMRMLSliceLogic::GetVolumeRASBox(volumeNode, xyz, center);
     for (int i = 0; i < 3; i++)
-    {
+      {
       xyz[i] *= 0.5;
-    }
+      }
 
     roiNode->SetXYZ(center);
     roiNode->SetRadiusXYZ(xyz);
+    }
+  else if (markupsROINode)
+    {
+    MRMLNodeModifyBlocker blocker(markupsROINode);
 
-    roiNode->EndModify(disabledModify);
-  }
+    double xyz[3] = { 0.0 };
+    double center[3] = { 0.0 };
+
+    vtkMRMLSliceLogic::GetVolumeRASBox(volumeNode, xyz, center);
+    for (int i = 0; i < 3; i++)
+      {
+      xyz[i] *= 0.5;
+      }
+
+    markupsROINode->GetROIToLocalMatrix()->Identity();
```

Comment:
Markups ROI should be rotated to match the volume axes.

Fitting ROI to a volume node could be a method in markups module logic (or maybe markups ROI node), as it can be useful for other modules, too.